### PR TITLE
WIP 10 sale line option

### PR DESCRIPTION
**Pull request by bealdav**
This module provide an options system for sale order line

bom

sale_line

### CODE REVIEW

**Comment by bealdav** [comment]
weird result flag on run bot because:


+=======================================
|  Tests summary:
|---------------------------------------
| test_server.py              [1;32mSuccess[0;m
+=======================================

File: sale_line_option/README.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+   :width: 800 px
+
+.. figure:: sale_line_option/static/description/sale_line.png
+   :alt: Sale line with options: no default option when Default Quantity is 0
+   :width: 800 px
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/167/8.0
```
Comment (JayVora-SerpentCS) [inline sale_line_option/README.rst:47]: 10.0

File: sale_line_option/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# coding: utf-8
+# © 2015 Akretion, Valentin CHEMIERE <valentin.chemiere@akretion.com>
+# © 2017 Akretion, David BEAL <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Sale Line Option',
+    'version': '10.0.0.0.1',
+    'author': 'Akretion, Odoo Community Association (OCA)',
+    'website': 'www.akretion.com',
```
Comment (JayVora-SerpentCS) [inline sale_line_option/__manifest__.py:10]: website -https://github.com/OCA/sale-workflow

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

File: sale_line_option/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    'depends': [
+        'sale_order_lot_mrp',
+        'onchange_helper',
+    ],
+    'data': [
+        'views/sale_view.xml',
+        'views/mrp_view.xml',
+        'views/stock_view.xml',
+        'views/install.xml',
+        'security/ir.model.access.csv',
```
Comment (nikul-serpentcs) [inline sale_line_option/__manifest__.py:21]: @bealdav `security` file always first.

File: sale_line_option/models/sale.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            else:
+                record.invalid_qty = False
+
+    @api.onchange('qty')
+    def onchange_qty(self):
+        for record in self:
+            if record.bom_line_id and record.opt_max_qty < record.qty:
+                max_val = record.qty
+                record.qty = record.opt_max_qty
+                return {'warning': {
```
Comment (nikul-serpentcs) [inline sale_line_option/models/sale.py:192]: here use `UserError`

File: sale_line_option/tests/test_change_option.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# coding: utf-8
+# © 2017 David BEAL @ Akretion <david.beal@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from openerp.tests.common import TransactionCase
```
Comment (nikul-serpentcs) [inline sale_line_option/tests/test_change_option.py]: `openerp` replace `odoo`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Comment by rousseldenis** [comment]
@bealdav  @sebastienbeau Is this still applicable ?

**Comment by bealdav** [comment]
I think not, but I let @sebastienbeau answer.

**Comment by sebastienbeau** [comment]
Hi we totally have refactored this module, from now the project is here : https://github.com/akretion/sale-configurator
When it will be ready in V14 we will propose the repo into the OCA

**Pull request closed without merging**